src/checkers/inference/dataflow/InferenceTransfer.java

```diff
@@ -59,10 +59,13 @@
 
     // Type variables will have two refinement variables (one for each bound).  This covers the
     // case where the correct, inferred RHS has no primary annotation
-    private Map<Tree, Pair<RefinementVariableSlot, RefinementVariableSlot>> createdTypeVarRefinementVariables = new HashMap<>();
+    private final Map<Tree, Pair<RefinementVariableSlot, RefinementVariableSlot>> createdTypeVarRefinementVariables = new HashMap<>();
```

Comment:
file an issue about `Tree` being the map key

Reply:
Filed #374